crates/server/src/api/fetch.rs

```diff
@@ -31,11 +29,25 @@ impl Config {
     }
 }
 
+#[derive(Debug, Error)]
+pub(crate) enum FetchApiError {
```

Comment:
Would it make sense to put errors into the `api` crate now or in the future?

Reply:
Let me see what I can do regarding moving these types into `api`; I think they belong there and it'd be nice if we could share the representation between the client and server (might require using a JSON response on error instead of text like we do now).